kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/ResourcesAdapter.kt

```diff
@@ -26,7 +28,7 @@ internal class ResourcesAdapter() : ListAdapter<Resource, ResourcesAdapter.ViewH
         position: Int,
     ) {
         val resource = getItem(position)
-        holder.bind(resource)
+        holder.bind(resource) { newResource -> onSwitchToggled(newResource) }
```

Comment:
Does this ultimately result in saving to the repo every time a view holder gets created? I tried to follow the function calls down, but I don't know how `bind` works here.

If you trace the save calls, it doesn't save just while loading or scrolling, right?

Reply:
kotlin syntax is super confusing! this is just passing a callback to the `bind`, which is later used as a callback from the button's actual callback.

It doesn't get called with each `bind`

Reply:
Okay I didn't realize it was for the button. I thought it was for the entire list item including the text and everything